File: ts/src/Counter.ts
import { HttpClient } from "./HttpClient";

export default class Counter {
  httpClient: HttpClient;
  name: String;
  ip: String;
  friendlyName: string;
  totalIn: any;
  totalOut: any;
  lastUpdate: any;
  serial: any;
  constructor(httpClient: HttpClient, name: String, ip: String) {
    this.httpClient = httpClient;
    this.name = name;
    this.ip = ip;
    this.update();
  }
  update() {
    try {
      const data = this.httpClient.fetch(
        "http://${ip}/people-counter/api/live.json"
      );
      this.totalIn = data.inAmount;
      this.totalOut = data.outAmount;
      this.friendlyName = data.name;
      this.lastUpdate = data.timestamp;
      this.serial = data.serial;
    } catch (e) {
      this.friendlyName = "error connectiong to ${name}";
    }
  }
}

File: elixir/lib/counter.ex
defmodule Counter do
  def new(httpClient, name, ip) do
    state = %{
      httpClient: httpClient,
      name: name,
      ip: ip
    }

    update(state)
  end

  def update(counter) do
    %{httpClient: httpClient, ip: ip} = counter
    data = HttpClient.fetch(httpClient, "http://#{ip}/people-counter/api/live.json")

    Map.merge(
      counter,
      %{
        total_in: data.inAmount,
        total_out: data.outAmount,
        friendly_name: data.name,
        last_update: data.timestamp,
        serial: data.serial
      }
    )
  end
end
